### ISSUE TITLE
What is the compute needed for GRPO for 7B R1-Distill model?

### ISSUE THREAD

**Issue by iamansinha** [huggingface/open-r1#100]
Anybody who has tried GRPO over any of the R1-Distill models: what is the minimum GPU compute requirement to run the training?
Let's say for R1-Distill-Qwen-7B ?

I am talking about this from the README:
---
### GRPO
```
accelerate launch --config_file configs/zero3.yaml src/open_r1/grpo.py \
    --output_dir DeepSeek-R1-Distill-Qwen-7B-GRPO \
    --model_name_or_path deepseek-ai/DeepSeek-R1-Distill-Qwen-7B \
    --dataset_name AI-MO/NuminaMath-TIR \
    --max_prompt_length 256 \
    --per_device_train_batch_size 1 \
    --gradient_accumulation_steps 16 \
    --logging_steps 10 \
    --bf16
```

**Comment by Some-random** [comment]
I only have A100 40G GPUs and I'm facing OOM errors. Does anyone have suggestions for me (e.g. change deepspeed config or using some sort of data parallelism)?

**Comment by qgallouedec** [comment]
You may want to use vllm for generation. Much faster. See https://huggingface.co/docs/trl/speeding_up_training

**Comment by LukasNel** [comment]
Try using peft perhaps?

**Comment by Some-random** [comment]
> I managed to get it running with 8xH100's on Modal, but it was slow going at 119s/it

Hi Lukas, do you have an estimate on how long does it take to finish the training?

**Comment by jhinpan** [comment]
I can provide one more data sample. Currently I am running the GRPO training on 6*A6000. And the speed will be like 195s/it. It will take about 49hrs to finish training. Don't know if I make some potential mistakes. But I think I should already had the correct usage of vllm to accelerate generation.

**Comment by ChewKokWah** [comment]
> I can provide one more data sample. Currently I am running the GRPO training on 6*A6000. And the speed will be like 195s/it. It will take about 49hrs to finish training. Don't know if I make some potential mistakes. But I think I should already had the correct usage of vllm to accelerate generation.

Is the 49hrs for full NuminaMath TIR dataset?

**Comment by zijianh4** [comment]
> I can provide one more data sample. Currently I am running the GRPO training on 6*A6000. And the speed will be like 195s/it. It will take about 49hrs to finish training. Don't know if I make some potential mistakes. But I think I should already had the correct usage of vllm to accelerate generation.

Hi @jhinpan ,
Thanks for your information. Is there any specific configuration that needs to be modified such as batch size on the GPUs with 40G memory? I would really appreciate it if you could share your configuration files and command. Thanks!

**Comment by thepowerfuldeez** [comment]
I've also shared a long list of results and observations here: https://x.com/iamgrigorev/status/1885391768666918928

VLLM is definitely a must

**Comment by Some-random** [comment]
Excellent observation @thepowerfuldeez ! Have you figured out a configuration that works for this repo? The [learning rate](https://github.com/huggingface/open-r1/blob/main/recipes/qwen/Qwen2.5-1.5B-Instruct/grpo/confg_full.yaml#L27) we have is 2e-05, which seems overly high?

**Comment by thepowerfuldeez** [comment]
Exactly, learning rate should be lower, 3e-6 worked great in my experiments.

**Comment by Some-random** [comment]
That's so cool! I'm trying to replicate the 7B+8K experiments documented [here](https://hkust-nlp.notion.site/simplerl-reason), their setup seems to be very similar to yours (you're both using 8K MATH data). Are you getting any success there (e.g. being able to outperform Qwen-2.5-Math-7B-Instruct using Qwen-2.5-Math-7B-Base + DRPO training on Math data)?

**Comment by Mryangkaitong** [comment]
> I only have A100 40G GPUs and I'm facing OOM errors. Does anyone have suggestions for me (e.g. change deepspeed config or using some sort of data parallelism)?

https://github.com/Mryangkaitong/deepseek-r1-gsm8k
